ion_schema_2_0/open_content/user_fields_declaration.isl

```diff
@@ -0,0 +1,655 @@
+$ion_schema_2_0
+
+$test::{
+  description: "user_fields declaration must be a non-null, unannotated struct",
+  invalid_schemas:[
+    (
+      $ion_schema_2_0
+      schema_header::{ user_fields: a_symbol }
+      schema_footer::{}
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{ user_fields: null.struct }
+      schema_footer::{}
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{ user_fields: [a, b, c] }
+      schema_footer::{}
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{ user_fields: foo::{} }
+      schema_footer::{}
+    ),
+  ]
+}
+
+$test::{
+  description: "user_fields declaration may not have unexpected fields",
+  invalid_schemas:[
+    (
+      $ion_schema_2_0
+      schema_header::{
+        type: [a, b, c],
```

Comment:
I thought it would be obvious because `foo` is commonly used for pseudocode and other "fake" code, and there is no `foo` keyword in ISL, but I can add a comment anyway.

```suggestion
        type: [a, b, c],
        // 'foo' is not one of the allowed fields in the user_fields struct
```